pkg/kubectl/cmd/set/set_env.go

```diff
@@ -471,7 +471,7 @@ func (o *EnvOptions) RunEnv() error {
 	for _, patch := range patches {
 		info := patch.Info
 		if patch.Err != nil {
-			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Object.GetObjectKind().GroupVersionKind().Kind, info.Name, patch.Err))
```

Comment:
kind is not the same as resource... this changes the output here

Reply:
Good point.

Reply:
Since we were panicing on --local before (hence no output to change under that code path), I can go ahead and leave info.Mapping.Resource as it was, and only output kind under --local. wdyt?

Reply:
That's a reasonable middle ground, eventually we can entirely skip that part of the message with `--local`

Reply:
I guess we can do the same thing we do for -o name without resource info... `lower(<kind>[.<group>])`

Reply:
Thanks, PR updated